web/src/features/panels/zone/ShareButton.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, ButtonProps } from 'components/Button';
+import { Toast, useToastReference } from 'components/Toast';
+import { isiOS, isMobile } from 'features/weather-layers/wind-layer/util';
+import { Share, Share2 } from 'lucide-react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { twMerge } from 'tailwind-merge';
+import { ShareType, trackShare } from 'utils/analytics';
+import { DEFAULT_ICON_SIZE } from 'utils/constants';
+
+interface ShareButtonProps
+  extends Omit<
+    ButtonProps,
+    'icon' | 'children' | 'href' | 'onClick' | 'backgroundClasses' | 'foregroundClasses'
+  > {
+  iconSize?: number;
+  shareUrl?: string;
+  showiOSIcon?: boolean;
+}
+const trackShareClick = trackShare(ShareType.SHARE);
+const DURATION = 3 * 1000;
+
+export function ShareButton({
+  iconSize = DEFAULT_ICON_SIZE,
+  shareUrl,
+  showiOSIcon = isiOS(),
```

Comment:
Nitpick but I think consistency is worth the ugliness in this case.
```suggestion
  showIosIcon = isIos(),
```

Reply:
I'm very pro releasing this on turning the feature flag on today 🚢

Reply:
Fixed - will merge it in when everything passes. Thanks for the review!